assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -633,21 +635,31 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
+
+      // Helper for checking webhook objects in the buildConfig spec and creating if missing.
+      function reuseOrCreateWebhook(type) {
+        var webhook = _.find($scope.buildConfig.spec.triggers, function(obj) { return obj.type === type})
```

Comment:
can you have more than one Github webhook or more than one Generic webhook?  we should make sure the editor isn't stripping out any additional ones

cc @bparees